### DISCUSSION TITLE
Looking for help with problem after updating from v1.10.0 to 1.15.4

### DISCUSSION THREAD

**Discussion by flodabo** [antonmedv/expr#475] [Q&A]
Hi,

im currently doing some work on a company internal nagios monitoring check.
it lets a user query whichever http json api they need and dynamically set the warning/critical status based on a user provided expression  and the response body as env.
i was going to update expr because of the date functions added.

however updating expr to the latest version caused a problem elsewhere and i was not able to solve it yet.
one API returns a property `count` in the response json that previously was used in an expression as such `count > 0`.
since the update to 1.15.4 this produces the following error:
```
invalid operation: > (mismatched types func(...interface {}) (interface {}, error) and int) (1:7)
| count > 0
```
the way i understand the message is that the new version now tries to parse count as the built-in `count()` function instead of a variable `count` in the env.

i have found neither a bug report nor any hint in the docs.
So my questions are:

* is this intended behavior or a bug?
* if this intended: is there a possibity to disable some built-in functions or another way to mitigate this behavior?
* in general: how to deal variables with the same name as functions? I tested some more functions(sum, all, mean) and observed the same behavior

**Comment by antonmedv** [comment]
new functions were introduced in expr to manage built-in functions:
	•	DisableAllBuiltins(): This function disables all built-in functions.
	•	DisableBuiltin(name string): This allows you to disable a specific built-in function by name.
	•	EnableBuiltin(): This re-enables a disabled built-in function
	
The $env variable is a map of all variables passed to the expression.

    foo.Name == $env["foo"].Name

**Reply by flodabo** [reply]
Thanks for your quick response.
using `$env['count']` instead of `count` works. seems like the best solution to me.

**Reply by antonmedv** [reply]
Or

    $env.count
